### PR TITLE
fix: Menu bugs (take two)

### PR DESCRIPTION
**Pull request by felixrieseberg**
This is exactly the same as #11385, but rebased on top of master. It didn't hang for me, so I'm reopening this PR to see how CI behaves now.

### CODE REVIEW

File: spec/api-menu-spec.js [hunk 1]
```diff
@@ -283,7 +283,52 @@ describe('Menu module', () => {
...
 
     it('returns immediately', () => {
-      menu.popup(w, {x: 100, y: 100, async: true})
+      const { browserWindow, x, y } = menu.popup(w, {x: 100, y: 101})
+
+      assert.equal(browserWindow, w)
+      assert.equal(x, 100)
+      assert.equal(y, 101)
+
+      menu.closePopup(w)
```
Comment (alexeykuzmin) [inline spec/api-menu-spec.js]: If the test fails the created popup won't be closed.
Was it the reason why the `afterEach` hook was timing out on Mac?

File: spec/api-menu-spec.js [hunk 2]
```diff
@@ -283,7 +283,52 @@ describe('Menu module', () => {
...
+    })
+
+    it('works without a given BrowserWindow and options', () => {
+      const { browserWindow, x, y } = menu.popup({x: 100, y: 101})
+
+      assert.equal(browserWindow.constructor.name, 'BrowserWindow')
+      assert.equal(x, 100)
+      assert.equal(y, 101)
+
+      menu.closePopup()
```
Comment (alexeykuzmin) [inline spec/api-menu-spec.js]: Shouldn't the `browserWindow` be passed to the `menu.closePopup()` call?

Comment (felixrieseberg) [reply spec/api-menu-spec.js]: The parameter is optional, so if you opened it without one (which uses the focused window), it should just close. Then again, I'm not sure how well getting the key window on CI works, so we could pass it here.

**Comment by alexeykuzmin** [comment]
All tests are green now! 🎉

**Comment by ckerr** [comment]
+1 on `afterEach`, that is a nice improvement.

I'm not totally convinced: if the timeout was caused by failure to close after a test failure, it seems like this addresses the timeout but not the test failure that came before it. Unless `menu.closePopup()` without a `w` was the problem there.

The tests aren't all green yet, I'm going to wait until they finish this time :smiling_imp:

**Comment by alexeykuzmin** [comment]
@ckerr The timeout was caused by a not closed popup, I've seen it right in the middle of my screen when I tried to run those tests. The tests themselves are fine.

> The tests aren't all green yet

All `Menu.popup` tests were green when I posted the previous message =)
I restarted CI jobs after because they failed on other, unrelated stuff.

**Comment by alexeykuzmin** [comment]
@ckerr, I think this fix should be cherry-picked to the `1-8-x` branch, right?

**Pull request merged**